src/openforms/appointments/api/views.py

```diff
@@ -52,10 +53,15 @@ class ProductsListView(ListMixin, APIView):
 
     def get_objects(self):
         plugin = get_plugin()
+        config = AppointmentsConfig().get_solo()
+        assert isinstance(config, AppointmentsConfig)
```

Comment:
In what situation would this not be a `AppointmentsConfig`?

Reply:
@SilviaAmAm It's always an AppointmentsConfig, but the assertion makes that clear to type checkers. Strictly speaking a `typing.cast` is sufficient here, but I prefer asserts, as assert will catch mistakes where cast does nothing during runtime, and gives you false confidence during static analysis.

Reply:
Really? An assert makes this clear to type checkers.

If so, then it's useful and should be patterniz d but in all other cases I think this is really redundant

Reply:
And to people too. :slightly_smiling_face:  From just reading this:
```python
    @classmethod
    def get_solo(cls):
        cache_name = getattr(settings, 'SOLO_CACHE', solo_settings.SOLO_CACHE)
        if not cache_name:
            obj, created = cls.objects.get_or_create(pk=cls.singleton_instance_id)
            return obj
        cache = get_cache(cache_name)
        cache_key = cls.get_cache_key()
        obj = cache.get(cache_key)
        if not obj:
            obj, created = cls.objects.get_or_create(pk=cls.singleton_instance_id)
            obj.set_to_cache()
        return obj
```
I can't infer the exact type of `obj`. It's the union of `Self` (because I know the contract of `cls.object.get_or_create`) and `Any` because we read "something" from some memory.  I think at `cache.get`, the library ought to make the assertion what we read from cache is an instance of `cls`, or, alternatively, `get_cache` should return a `Mapping[..., Self]` with the return type of `cls.get_cache_key()` at the ellipsis (probably `str`). In principle the cache is mutable by anything; and assertions guarantee our assumptions.

BTW this is exactly the use for the `assert` keyword: asserting the obvious, that ought to be true and may be optimized out during runtime with `python -O`